### ISSUE TITLE
How to deal with data sets with different number of channels?

### ISSUE THREAD

**Issue by Leonssx** [935963004/LaBraM#23]
Thank you very much for the code. I would like to ask you, you mentioned in the code that you need to provide different time windows to ensure that the sequence length of each data set is equal to 256, but for data sets with the number of channels less than 64/32, do we need to complete it to 64/32? 32?

**Comment by 935963004** [comment]
No, just ensure the sequence length not exceed 256.

**Comment by Leonssx** [comment]
Okay, thank you very much for your answer. I have another question for you: when training vqsnp, involving training set and test set, how should we divide the training set and test set, or use that data as Training set and test set? Thanks again

**Comment by 935963004** [comment]
The test set is divided to ensure the generalizability of vqnsq. In my experiment, the loss on test set can converge to a low value thus the generalizability is proven. It is okay if you don't want to divide a test set.

**Comment by Leonssx** [comment]
When I use the processed data for vasp training, a warning of WARNING:root:NaN or Inf found in input tensor. will appear. The grad_norm during the training process will also appear with an (inf) mark. Do you know the reason? (Data preprocessing is done strictly in accordance with the requirements）
![Uploading 图片1.png…]()
![Uploading 图片1.png…]()
![Uploading 图片1.png…]()
![Uploading 图片1.png…]()

**Comment by 935963004** [comment]
This might be due to some outlier data. If the training is not influenced and the loss can converge, you may just ignore it.

**Comment by Leonssx** [comment]
OK, thank you very much for your reply, I will continue to try

**Comment by Gorsonpy** [comment]
I also encountered this problem, how did you solve it in the end?
